src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/5.diff.sql

```diff
@@ -0,0 +1,1613 @@
+/****************************************************************************************
+Guidelines to create migration scripts - https://github.com/microsoft/healthcare-shared-components/tree/master/src/Microsoft.Health.SqlServer/SqlSchemaScriptsGuidelines.md
+******************************************************************************************/
+SET XACT_ABORT ON
+
+BEGIN TRANSACTION
+
+---BEGIN - CREATE/ALTER Statements----------
+
+IF NOT EXISTS (
+    SELECT *
+    FROM sys.tables
+    WHERE name = 'Partition')
+BEGIN
+    CREATE TABLE dbo.Partition (
+        PartitionKey            BIGINT                 NOT NULL, --PK
+        PartitionId             VARCHAR(64)            NOT NULL    DEFAULT 'Microsoft.Default',
+        --audit columns
+        CreatedDate             DATETIME2(7)           NOT NULL
+    ) WITH (DATA_COMPRESSION = PAGE)
+
+    -- Inserting Default values since we are adding PartitionKey as NOT NULL columns in all the tables
+    INSERT INTO dbo.Partition (PartitionKey, PartitionId, CreatedDate)
+    VALUES (1, 'Microsoft.Default', SYSUTCDATETIME())
+END
+
+IF NOT EXISTS (
+    SELECT *
+    FROM sys.indexes
+    WHERE name='IXC_Partition' AND object_id = OBJECT_ID('dbo.Partition'))
+BEGIN
+    CREATE UNIQUE CLUSTERED INDEX IXC_Partition ON dbo.Partition
+    (
+        PartitionKey
+    )
+END
+GO
+
+IF NOT EXISTS (
+    SELECT *
+    FROM sys.indexes
+    WHERE name='IXC_Partition_PartitionKey_PartitionId' AND object_id = OBJECT_ID('dbo.Partition'))
+BEGIN
+    CREATE UNIQUE NONCLUSTERED INDEX IXC_Partition_PartitionKey_PartitionId ON dbo.Partition
+    (
+        PartitionKey,
```

Comment:
Both the indexes are same, You are filtering on PartitionKey first. 

Don't you need a index to find the Key based on used specified PartitionId

Reply:
You can also use to enforce Unique PartitionId

Reply:
Changing the index on `PartitionName` to only index that column.